errors.go

```diff
@@ -77,13 +77,13 @@ func extractMessage(resp *http.Response) string {
 			return fmt.Sprintf("<invalid json in response body: %v>", err)
 		}
 
-		keys := make([]string, len(doc), 0)
```

Comment:
forgot to say, this would've crashed with a non-empty doc map, right? could add a test case for it